netket/graph/lattice.py

```diff
@@ -176,7 +230,10 @@ def __init__(
                 for atom_coord in atoms_coord
             ]
         )
-        if atoms_coord_fractional.min() < 0 or atoms_coord_fractional.max() >= 1:
+        if (
+            atoms_coord_fractional.min() < -cutoff_tol
```

Comment:
I don't undestand this.
 the check before was `atoms_coords_fractional.min < 0` equivalent to saying if `atom_cpoords_frac` is negative.
Now you are checking that it should be larger than a cutoff.

Why is that? In general floating point math has rounding errors but does not get the sign wrong easily. 
Did you have a case that would break otherwise? (if so, is it in the tests?)

Reply:
Floating points regularly get the sign wrong. For example 1.0 - 1.0 is sometimes positive and sometimes negative. The old code broke any lattice where the atoms were specified to be on the boundary of the unit cell. 

I don’t get the point of this line tbh, the code checks for duplicate atoms somewhere else anyway.